<?xml version="1.0"?>
<!--

This launch file was intended for stand-alone use
It starts the following nodes:
- The actual controller (cpp executable)
- A state estimation node (cpp executable) , deriving the velocity from the position information from the OptiTrack

Launch this node after running an instance of the ardrone_autonomy package, and an instance of the mocap_optitrack package to get the quadcopter to the setpoint configured below.

  -->

<launch>
  <node name="control" pkg="ram" type="controller" output="screen">
    <param name="setpoint_x" value="0" />            <!-- Setpoint in world pose frame -->
    <param name="setpoint_y" value="0" />            <!-- Setpoint in world pose frame -->
    <param name="setpoint_z" value="1.4" />          <!-- Setpoint in world pose frame -->
    <param name="setpoint_yaw" value="0" />          <!-- Setpoint in world pose frame -->
    <param name="gain_p_z" value="1.2" />            <!-- Translational P gain for vertical speed -->
    <param name="gain_d_z" value="0" />              <!-- Translational D gain for vertical speed -->
    <param name="gain_p_translational" value="0.12" />    <!-- Translational P gain for lateral motion -->
    <param name="gain_d_translational" value="0.08" />    <!-- Translational D gain for lateral motion -->
    <param name="gain_p_rotational" value="0.9" />        <!-- Rotational P gain -->
    <param name="gain_d_rotational" value="0.2" />        <!-- Rotational D gain -->
    <param name="simulation" value="false" />             <!-- True if data is to be read from Gazebo (OLD) -->
    <param name="hovermode" value="false" />              <!-- Enable hovermode if error in all directions is smaller than the hover_treshold -->
    <param name="hover_treshold" value="0.05" />          <!-- Hover treshold -->
    <param name="K" value="5" />                    <!-- OLD -->
    <param name="uncertainty_band" value="0.08" />  <!-- OLD -->
    <param name="pose_memory" value="30" />         <!-- Number of poses to remember for error calculation -->
    <param name="publish_rate" value="35" />        <!-- Update frequency in HZ -->
    <param name="gain_velocity" value="-0.3" />     <!-- Standard gain for velocity damping -->
    <param name="velocity_damping" value="false" /> <!-- Enable or disable absolute velocity damping -->
    <param name="i_action" value="false" />         <!-- Enable or disable integral action on start. There is absolutely no reason to enable this. -->
    <param name="gain_i" value="0.01" />            <!-- Integral action gain (equal for lateral actions) -->
  </node>

  <node pkg="ram" type="state_estimation" name="state_estimation" output="screen">
    <param name="K" value="70" />                   <!-- OLD -->
    <param name="uncertainty_band" value="0.01" />  <!-- OLD -->
    <param name="pose_memory" value="25" />         <!-- Number of poses to remember for velocity calculation -->
  </node>
</launch>